### PR TITLE
feat!: Added advanced logging configuration. Bump version of AWS provider to 5.32

### LINKED ISSUE #530: feat: Add support for logging_config
## Is your request related to a new offering from AWS?
<https://aws.amazon.com/blogs/compute/introducing-advanced-logging-controls-for-aws-lambda-functions/>

Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)?

* Yes ✅: [v5.32.0](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.32.0)

## Describe the solution you'd like.
support newly introduced `logging_config` for lambda function argument.

### PR DESCRIPTION
**Pull request by JoshuaLicense**
## Description
Adds `logging_config` block for enabling [advanced logging](https://aws.amazon.com/blogs/compute/introducing-advanced-logging-controls-for-aws-lambda-functions/).

## Motivation and Context
Closes #530 

## Breaking Changes
No

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -112,6 +112,16 @@ resource "aws_lambda_function" "this" {
     }
   }
 
+  dynamic "logging_config" {
+    for_each = var.logging_config_log_format != null ? [true] : []
+    content {
+      log_format            = var.logging_config_log_format
```
Comment (antonbabenko) [inline main.tf:118]:
```suggestion
      log_format            = var.logging_log_format
```

Let's drop `config_` from variable names like we do in other places.

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Could you please update this module and handle the cases when the user specifies custom log group then default log group should not be created?

Maybe there are other cases that should be considered. I would appreciate if you can think about this and update this PR.

Also, maybe we need to update the log group in the outputs of the module if user specifies custom log group.

**Comment by JoshuaLicense** [comment]
> Could you please update this module and handle the cases when the user specifies custom log group then default log group should not be created?
>
> Maybe there are other cases that should be considered. I would appreciate if you can think about this and update this PR.

My bad, thanks for the review. I've pushed some tweaks that handle the different log group scenarios better.
| `use_existing_cloudwatch_log_group` | `logging_log_group` | `aws_cloudwatch_log_group` created |
| --- | --- | --- |
| `true` | `null` | ❌ retrieved from data source: `"/aws/lambda/${var.lambda_at_edge ? "us-east-1." : ""}${var.function_name}"` |
| `true` | _populated_ | ❌ retrieved from data source: `var.logging_log_group` |
| `false` | `null` | ✅ 
| `false` | _populated_ | ❌ retrieved from data source: `var.logging_log_group`

> Also, maybe we need to update the log group in the outputs of the module if user specifies custom log group.

Existing data source is re-used so outputs should update appropriately.

File: main.tf [hunk 2]
```diff
@@ -112,6 +112,16 @@ resource "aws_lambda_function" "this" {
     }
   }
 
+  dynamic "logging_config" {
+    for_each = var.logging_log_format != null ? [true] : []
```
[the remark this answers is not in the document]
Comment (JoshuaLicense) [reply main.tf:116]: `log_format` is required, it cannot be passed as `null`. 

https://github.com/hashicorp/terraform-provider-aws/blob/85b0843d0b3963876fcff0c5f55259239de8a558/internal/service/lambda/function.go#L250

Am I misunderstanding?

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 7.0.0 :tada:

**Comment by AndrewCharlesHay** [comment]
This causes breaks in Gov Cloud see #539

**Pull request merged**